pservice/pdo/pservice/utility/ias_client.py

```diff
@@ -16,53 +16,57 @@
 Provide rest api helper functions for communicating with IAS.
 """
 
-from urllib.parse import urljoin
-
 import requests
-
 import sys
-def printf(format, *args):
-    sys.stdout.write(format % args)
+
+import logging
+logger = logging.getLogger(__name__)
+
 
 class IasClient(object):
     """
     Provide rest api helper functions for communicating with IAS.
     """
 
     def __init__(self, **kwargs):
+        logger.info("IAS settings:")
         self._proxies = {}
         if "HttpsProxy" in kwargs:
             self._proxies["https"] = kwargs["HttpsProxy"]
+            logger.info("Proxy: " + self._proxies["https"])
         if "Spid" in kwargs:
```

Comment:
in general... string concatenation happens whether the message is printed or not... i believe printf formatting happens only when the logger message is being displayed.

Reply:
this is what eservice/pdo/eservice already had, i just made sure they are identical so a diff could reveal whether i forgot to change something in one which i did in the other.

Reply:
I've been trying to convince others to avoid string concatenations. I think @bvavala started it. Better practice is to just use printf style (which is what logger module was designed for).